ffi/java-idsdk/src/main/java/com/iotics/sdk/identity/experimental/ResolverClient.java

```diff
@@ -0,0 +1,109 @@
+package com.iotics.sdk.identity.experimental;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Base64;
+import java.util.Objects;
+
+public class ResolverClient {
+    private final URL base;
+    private final int DEFAULT_TIMEOUT = 2000;
+    private final OkHttpClient client;
+
+    public ResolverClient(URL base) {
+        this.base = base;
+        this.client = new OkHttpClient();
+    }
+
+    public final class Result {
+        private final String content;
+        private final String contentType;
+        private final boolean isErr;
+
+        public Result(String content, String contentType, boolean isErr) {
+            this.content = content;
+            this.contentType = contentType;
+            this.isErr = isErr;
+        }
+
+        @Override
+        public String toString() {
+            return "Result{" +
+                    "content='" + content + '\'' +
+                    ", contentType='" + contentType + '\'' +
+                    ", isErr=" + isErr +
+                    '}';
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Result result = (Result) o;
+            return isErr == result.isErr && Objects.equals(content, result.content) && Objects.equals(contentType, result.contentType);
```

Comment:
Why do we need to be able to compare two results.
Also: If it does need to be compared - surely it's exactly the same if `result.content` match even if `contentType` is different - same data, no?

Reply:
this is object equality.
since i had to downgrade to java 11 it's always best practice to add an impl of equals.

in java 16 this would have been a record (instead of a class) which would have behaved in the same manner.